tests/rptest/tests/follower_fetching_test.py

```diff
@@ -0,0 +1,148 @@
+                    total_bytes += sample.value
+        return total_bytes
+
+    def create_consumer(self, topic, rack=None):
+
+        properties = {}
+        if rack:
+            properties['client.rack'] = rack
+        return KafkaCliConsumer(
+            self.test_context,
+            self.redpanda,
+            topic=topic,
+            group='test-gr-1',
+            from_beginning=True,
+            consumer_properties=properties,
+            formatter_properties={
+                'print.value': 'false',
+                'print.key': 'false',
+                'print.partition': 'true',
+                'print.offset': 'true',
+            },
+        )
+
+    @cluster(num_nodes=5)
+    @matrix(read_from_object_store=[True, False])
+    def test_basic_follower_fetching(self, read_from_object_store):
+        rack_layout_str = "ABC"
+        rack_layout = [str(i) for i in rack_layout_str]
+
+        for ix, node in enumerate(self.redpanda.nodes):
+            extra_node_conf = {
+                # We're introducing two racks, small and large.
+                # The small rack has only one node and the
+                # large one has four nodes.
+                'rack': rack_layout[ix],
+                # This parameter enables rack awareness
+                'enable_rack_awareness': True,
+            }
+            self.redpanda.set_extra_node_conf(node, extra_node_conf)
+
+        self.redpanda.start()
+        topic = TopicSpec(partition_count=1, replication_factor=3)
+
+        self.client().create_topic(topic)
+
+        self.produce(topic.name)
+        self.logger.info(f"Producing to {topic.name} finished")
+        if read_from_object_store:
+            RpkTool(self.redpanda).alter_topic_config(
+                topic.name,
+                TopicSpec.PROPERTY_RETENTION_LOCAL_TARGET_BYTES,
+                self.local_retention,
+            )
+            wait_for_local_storage_truncate(self.redpanda,
+                                            topic.name,
+                                            target_bytes=self.local_retention)
+        node_idx = random.randint(0, 2)
+        consumer_rack = rack_layout_str[node_idx]
+        self.logger.info(f"Using consumer with {consumer_rack}")
+        consumer = self.create_consumer(topic.name, rack=consumer_rack)
+        consumer.start()
+        consumer.wait_for_messages(1000)
+
+        preferred_replica = self.redpanda.nodes[node_idx]
+        fetched_per_node = {}
+        self.logger.info(
+            f"preferred replica {preferred_replica.account.hostname}:{self.redpanda.node_id(preferred_replica)} in rack {consumer_rack}"
+        )
+        for n in self.redpanda.nodes:
+            fetched_per_node[n] = self.get_fetch_bytes(n, topic.name)
+            self.logger.info(
+                f"fetched {fetched_per_node[n]} bytes from node {n.account.hostname}:{self.redpanda.node_id(n)}"
+            )
+
+        assert fetched_per_node[preferred_replica] > 0
```

Comment:
it seems like the test isn't strong without

1. checking a property about the bytes fetched on all nodes (not just preferred_replica)
2. that the preferred replica doesn't happen to be the a partition leader

Could we repeat this entire cycle in a loop with some random choices?

Reply:
I agree with the first point. The second one is actually not true in the general case as it may happen that the closest replica is a leader.